### PR TITLE
Add doc to sense_to_set

### PR DESCRIPTION
**Pull request by blegat**
See https://github.com/JuliaOpt/JuMP.jl/issues/1825

### CODE REVIEW

File: docs/src/extensions.md [hunk 1]
```diff
@@ -51,6 +51,14 @@ The [`@constraint`](@ref) macro always calls the same three functions:
 Adding methods to these functions is the recommended way to extend the
 [`@constraint`](@ref) macro.
 
+### Adding `parse_constraint` methods
+
+TODO
```
Comment (mlubin) [inline docs/src/extensions.md]: Please avoid leaving visible TODOs in the documentation now that we're preparing for the release. You can use the `@meta` tag and say
```
# TODO(Benoit): Introduce the `set_to_sense` docstring.
```
instead (e.g., at the top of https://raw.githubusercontent.com/JuliaOpt/JuMP.jl/master/docs/src/index.md).

Comment (blegat) [reply docs/src/extensions.md]: Good point, I forgot about this

File: src/macros.jl [hunk 2]
```diff
@@ -358,6 +358,47 @@ end
...
+    sense_to_set(_error::Function, ::Val{sense_symbol})
+
+Converts a sense symbol to a set `set` such that
+`@constraint(model, func sense_symbol 0) is equivalent to
+`@constraint(model, func in set)` for any `func::AbstractJuMPScalar`.
+
+## Example
+
+Once a custom set is defined you can directly create a JuMP constraint with it:
+```jldoctest sense_to_set; setup = :(using JuMP)
```
Comment (mlubin) [inline src/macros.jl]: Nit: Why is `using JuMP` needed for the doctest while `MOIU` doesn't need to be defined?

File: docs/src/extensions.md [hunk 3]
```diff
@@ -51,6 +51,17 @@ The [`@constraint`](@ref) macro always calls the same three functions:
 Adding methods to these functions is the recommended way to extend the
 [`@constraint`](@ref) macro.
 
+### Adding `parse_constraint` methods
+
+```@meta
+# TODO(Benoît): Details how `parse_constraint` works and shows how
```
Comment (mlubin) [inline docs/src/extensions.md]: Grammar nit: "Detail how `parse_constraint` works and show how `sense_to_set` fits into the picture."

File: src/macros.jl (hunk 2 above)
Comment (blegat) [reply src/macros.jl]: `MOIU` is exported by JuMP

**Pull request merged**